relay-pii/src/processor.rs

```diff
@@ -16,9 +16,112 @@ use relay_protocol::{Annotated, Array, Meta, Remark, RemarkType, Value};
+    state: &ProcessingState<'_>,
+    compiled_config: &CompiledPiiConfig,
+) -> ProcessingResult {
+    let patterns = [
+        SelectedPattern {
+            regex: &EMAIL_REGEX,
+            replacement: "[Email]",
+            remark_suffix: "email",
+        },
+        SelectedPattern {
+            regex: &CREDITCARD_REGEX,
+            replacement: "[CreditCard]",
+            remark_suffix: "creditcard",
+        },
+        SelectedPattern {
+            regex: &IBAN_REGEX,
+            replacement: "[IBAN]",
+            remark_suffix: "iban",
+        },
+        SelectedPattern {
+            regex: &US_SSN_REGEX,
+            replacement: "[SSN]",
+            remark_suffix: "ssn",
+        },
+    ];
+
+    let original_length = value.len();
+    let mut applied_patterns = Vec::with_capacity(patterns.len());
+
+    for pattern in &patterns {
+        let new_value = pattern.regex.replace_all(value, pattern.replacement);
+        if matches!(new_value, std::borrow::Cow::Owned(_)) {
+            *value = new_value.into_owned();
+            applied_patterns.push(pattern.remark_suffix);
+        }
+    }
+
+    // Add all remarks for applied patterns and set original length if needed.
+    if !applied_patterns.is_empty() {
+        for suffix in applied_patterns {
+            meta.add_remark(Remark::new(
+                RemarkType::Substituted,
+                format!("logentry.formatted:{suffix}"),
+            ));
+        }
+
+        // Set original length if the length changed.
+        if original_length != value.len() {
+            meta.set_original_length(Some(original_length));
+        }
+    }
+
+    // Apply user-configured PII rules, but avoid rules that would completely filter logentry.formatted.
+    // We skip @anything rules and only apply pattern-based rules to preserve context.
+    for (selector, rules) in compiled_config.applications.iter() {
+        if selector.matches_path(&state.path()) {
+            for rule in rules {
+                // Skip @anything rules for logentry.formatted to avoid complete filtering.
+                if rule.ty == RuleType::Anything {
+                    continue;
+                }
+
+                // Also skip rules with Default or Remove redaction that could delete everything.
+                if matches!(
+                    rule.redaction,
+                    Redaction::Default | Redaction::Remove | Redaction::Mask
+                ) {
+                    continue;
+                }
+
+                // Skip RedactPair rules that would replace the entire value with "[Filtered]".
+                // This prevents sensitive_fields configuration from completely filtering logentry.formatted.
+                if matches!(rule.ty, RuleType::RedactPair(_)) {
+                    if let Redaction::Replace(replace_redaction) = &rule.redaction {
+                        if replace_redaction.text == "[Filtered]" {
+                            continue;
```

Comment:
Maybe nicer with a let chain instead of nested ifs.
```suggestion
                if matches!(rule.ty, RuleType::RedactPair(_))
                    && let Redaction::Replace(replace_redaction) = &rule.redaction
                    && replace_redaction.text == "[Filtered]"
                {
                    continue;
```